splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/CrossJoinStrategy.java

```diff
@@ -231,6 +232,11 @@ public void estimateCost(Optimizable innerTable,
 
         // Only use cross join when it is inner join and run on Spark
         innerCost.setBase(innerCost.cloneMe());
+        double estimatedRowCount = innerCost.getEstimatedRowCount();
+        SConfiguration configuration=EngineDriver.driver().getConfiguration();
+        long rowCountThreshold = configuration.getBroadcastRegionRowThreshold();
+        broadcastInner = estimatedRowCount < rowCountThreshold;
```

Comment:
I just realize that we share the same CrossJoinStrategy for all cross joins in the same query, so set the braodcastInner flag in one cross join will be applied to all cross joins in the query, which is not right.

Reply:
As we discussed on the call, I removed the field from CrossJoinStrategy class and went around other engineering restrictions. Tests are also improved according to previous comments.